miniapps/autodiff/admfem.hpp

```diff
@@ -438,7 +438,7 @@ namespace mfem
 namespace ad
 {
 /// MFEM native forward AD-type
-typedef FDualNumber<double> ADFloatType;
+typedef internal::dual<double, double> ADFloatType;
```

Comment:
it supports different value and gradient types, making it composable with `tensor`, so that you can differentiate tensor-valued functions in a single invocation. See, for example:

https://serac.readthedocs.io/en/latest/sphinx/dev_guide/tensor_dual.html#using-tensor-and-dual-together